### PR TITLE
Eliminate hand-written C code for Lwt jobs bindings; simplify error handling

### PR DESCRIPTION
**Pull request by yallop**
This pull request removes the hand-written C code for the [readdir Lwt binding][readdir-binding], replacing it with code generated by the [new Lwt jobs support in ctypes 0.6](https://github.com/ocamllabs/ocaml-ctypes/pull/391) and additionally replaces the implementations of `opendir` and `closedir` with versions based on generated code.

Additionally, d269078 uses the [new errno support in ctypes 0.6](https://github.com/ocamllabs/ocaml-ctypes/pull/392) to simplify error handling in the `unix-dirent.unix` subpackage.

[readdir-binding]: https://github.com/dsheets/ocaml-unix-dirent/blob/b2bc93537f6a1b56012df688fbb94865f5e44288/lwt/unix_dirent_lwt_stubs.c
[lwt-jobs-support]: https://github.com/ocamllabs/ocaml-ctypes/pull/391

### CODE REVIEW

File: myocamlbuild.ml [hunk 1]
```diff
@@ -43,7 +43,22 @@ dispatch begin
...
       ~deps: ["lib_gen/%_bindgen.byte"]
       (fun env build ->
-        Cmd (A(env "lib_gen/%_bindgen.byte")));
+         Cmd (S[A(env "lib_gen/%_bindgen.byte");
+                A"--c-file";
+                A(env "unix/%_stubs.c");
+                A"--ml-file";
+                A(env "unix/%_generated.ml")]));
+ 
+    rule "cstubs: lwt/x_bindings.ml -> x_stubs.c, x_generated.ml"
```
Comment (dsheets) [inline myocamlbuild.ml]: This synopsis should have `lwt` in it?

Comment (yallop) [inline myocamlbuild.ml]: Do you mean you'd like to see `lwt` on the rhs of the `->`?

File: lib_gen/unix_dirent_bindings.ml [hunk 2]
```diff
@@ -40,12 +40,12 @@ let dir_handle = Ctypes.(
...
-    F.(foreign "unix_dirent_opendir" (string @-> returning dir_handle))
+    F.(foreign "opendir" (string @-> returning dir_handle))
 
   let readdir =
-    F.(foreign "unix_dirent_readdir" (dir_handle @-> returning (ptr_opt dirent)))
+    F.(foreign "readdir" (dir_handle @-> returning (ptr_opt dirent)))
 
   let closedir =
-    F.(foreign "unix_dirent_closedir" (dir_handle @-> returning int))
+    F.(foreign "closedir" (dir_handle @-> returning int))
```
Comment (dsheets) [inline lib_gen/unix_dirent_bindings.ml]: I think this removes the runtime unlocking?

Comment (dsheets) [inline lib_gen/unix_dirent_bindings.ml]: Oh, never mind I see your concat trick.

File: myocamlbuild.ml [hunk 3]
```diff
@@ -43,7 +43,22 @@ dispatch begin
...
-        Cmd (A(env "lib_gen/%_bindgen.byte")));
+         Cmd (S[A(env "lib_gen/%_bindgen.byte");
+                A"--c-file";
+                A(env "unix/%_stubs.c");
+                A"--ml-file";
+                A(env "unix/%_generated.ml")]));
+ 
+    rule "cstubs: lwt/x_bindings.ml -> lwt/x_stubs.c, lwt/x_generated.ml"
+      ~prods:["lwt/%_lwt_stubs.c"; "lwt/%_lwt_generated.ml"]
+      ~deps: ["lib_gen/%_bindgen.byte"]
```
Comment (dsheets) [inline myocamlbuild.ml]: This is still not quite right (pattern doesn't match `prods` and `deps` but it doesn't *really* matter and I'd like to get the change in.

**Pull request merged**